packages/material-ui/src/Select/Select.js

```diff
@@ -125,8 +125,11 @@ Select.propTypes = {
   /**
    * If `true`, a value is displayed even if no items are selected.
    *
-   * In order to display a meaningful value, a function should be passed to the `renderValue` prop which returns the value to be displayed when no items are selected.
-   * You can only use it when the `native` prop is `false` (default).
+   * In order to display a meaningful value, a function can be passed to the `renderValue` prop which
+   * returns the value to be displayed when no items are selected.
+   *
+   * ⚠️ When using this prop, make sure the label doesn't conflict with the empty displayed value.
```

Comment:
What does that mean? Could you give an example how _the_ label would conflict with the empty displayed value?

Reply:
?
```suggestion
   * ⚠️ When using this prop, make sure the label doesn't overlap with the empty displayed value.
```

Reply:
I think so. What techniques do we have to prevent it? Would be clearer to include these as well. Given the documentation I do not know how to make the codesandbox in the referenced issue "work".

Reply:
```diff
      <FormControl variant="outlined" className={classes.formControl}>
-       <InputLabel id="demo-simple-select-outlined-label">Age</InputLabel>
        <Select
          labelId="demo-simple-select-outlined-label"
          id="demo-simple-select-outlined"
          value={age}
          onChange={handleChange}
-         label="Age"
+         aria-label="Age"
          displayEmpty
        >
          <MenuItem value="">
            <em>None</em>
          </MenuItem>
          <MenuItem value={10}>Ten</MenuItem>
          <MenuItem value={20}>Twenty</MenuItem>
          <MenuItem value={30}>Thirty</MenuItem>
        </Select>
      </FormControl>
```

or 

```
      <FormControl variant="outlined" className={classes.formControl}>
-       <InputLabel id="demo-simple-select-outlined-label">Age</InputLabel>
+       <InputLabel shrink id="demo-simple-select-outlined-label">Age</InputLabel>
        <Select
          labelId="demo-simple-select-outlined-label"
          id="demo-simple-select-outlined"
          value={age}
          onChange={handleChange}
          label="Age"
          displayEmpty
        >
          <MenuItem value="">
            <em>None</em>
          </MenuItem>
          <MenuItem value={10}>Ten</MenuItem>
          <MenuItem value={20}>Twenty</MenuItem>
          <MenuItem value={30}>Thirty</MenuItem>
        </Select>
      </FormControl>
```

Reply:
This is how it looks using the second technique:
![2020-09-15 14_56_05-Material demo (forked) - CodeSandbox](https://user-images.githubusercontent.com/3032687/93207446-a7f69700-f763-11ea-9abe-af0b346cc27f.png)

Reply:
Need to also tell the outlined that it's shrunk.

Reply:
This looks to me more like a bug than a missing/incomplete documentation. If I were you I would drop this PR and concentrate more on how to make it actually work in the context of a filled/outlined variant. I don't think any of the suggested techniques has an acceptable outcome (removed label vs strike-through label).

Reply:
I'm not aware of any alternative that is viable (worth the tradeoff). The components are interdependent, they have to have a state that is compatible between each other.